### PR TITLE
app-misc/piper: New version

### PR DESCRIPTION
**Pull request by BratishkaErik**
Package-Manager: Portage-3.0.30, Repoman-3.0.3
Signed-off-by: BratishkaErik <bratishkaerik@getgoogleoff.me>

### CODE REVIEW

File: app-misc/piper/piper-0.5.1-r2.ebuild [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{8..9} )
+PYTHON_COMPAT=( python3_{8..10} )
```
Comment (kwhat) [inline app-misc/piper/piper-0.5.1-r2.ebuild:6]: So python 3.10 doesn't work seem to work:

    # ~ $ piper 
    Traceback (most recent call last):
      File "/usr/bin/piper", line 36, in <module>
        from piper.application import Application
    ModuleNotFoundError: No module named 'piper'

File: app-misc/piper/piper-0.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..10} )
```
Comment (kwhat) [inline app-misc/piper/piper-0.6.ebuild:6]: Same issue with python 3.10

File: app-misc/piper/piper-0.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+"
+DEPEND="
+	${RDEPEND}
+	dev-libs/libevdev
+	virtual/libudev
+"
+
+src_configure() {
+	python_setup
+	# Remove linting test that is broken upstream.
```
Comment (kwhat) [inline app-misc/piper/piper-0.6.ebuild:45]: Should we be disabling all tests?  I haven't looked into the tests in detail but there are a few others in addition to the failing flake8 test.

**Review by kwhat — CHANGES_REQUESTED** [review]

File: app-misc/piper/piper-0.5.1-r2.ebuild (hunk 1 above)
Comment (kwhat) [reply app-misc/piper/piper-0.5.1-r2.ebuild:6]: This looks like it is broken in meson:

    Program python3 (lxml, evdev, cairo, gi) found: YES (/usr/bin/python3.9) modules: lxml, evdev, cairo, gi
    Program python3 found: YES (/var/tmp/portage/app-misc/piper-0.6/temp/python3.10/bin/python3)

We maybe able to work around the problem with [0.62](https://github.com/mesonbuild/meson/pull/9788/files#diff-459a5de51a78f1bdaaa5b3032033df38a438d0c69fc66180d4a54a99c7e50714) but I am out of ideas on how to fix this for the current version.

Comment (kwhat) [reply app-misc/piper/piper-0.5.1-r2.ebuild:6]: Never mind, I'm an idiot.  This works just fine, I built all the dependencies with python 3_10 and forgot to rebuild meson.

File: app-misc/piper/piper-0.5.1-r2.ebuild [hunk 4]
```diff
@@ -1,11 +1,11 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
```
[the remark this answers is not in the document]
Comment (BratishkaErik) [reply app-misc/piper/piper-0.5.1-r2.ebuild]: done

File: app-misc/piper/piper-0.5.1-r2.ebuild (hunk 1 above)
Comment (BratishkaErik) [reply app-misc/piper/piper-0.5.1-r2.ebuild:6]: don't worry, it happens to all :)

**Review by juippis — COMMENTED** [review]
Could you perhaps look into bumping PYTHON_COMPAT for libratbag as well in this PR? I'm getting merge conflicts since it only supports up to  python-3.9 currently.

**Comment by BratishkaErik** [comment]
oops oops oops

File: app-misc/piper/piper-0.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	${PYTHON_DEPS}
+	virtual/pkgconfig
+	test? (
+		$(python_gen_cond_dep '
+			dev-python/flake8[${PYTHON_USEDEP}]
+		')
```
Comment (juippis) [inline app-misc/piper/piper-0.6.ebuild:27]: We try not to, 
https://projects.gentoo.org/python/guide/distutils.html?highlight=flake#undesirable-test-dependencies

and the test phase fails anyway.

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**